photon-client/src/main/scala/com/linkedin/photon/ml/io/scopt/game/ScoptGameTrainingParametersParser.scala

```diff
@@ -0,0 +1,190 @@
+          s"optional args: ${ScoptParserHelpers.formatArgs(ScoptParserHelpers.COORDINATE_CONFIG_OPTIONAL_ARGS)}"),
+        isRequired = true),
+
+      // Coordinate Update Sequence
+      ScoptParameter[Seq[CoordinateId], Seq[CoordinateId]](
+        GameTrainingDriver.coordinateUpdateSequence,
+        print = ScoptParserHelpers.iterableToString,
+        usageText = "<coordinate1>,<coordinate2>,...",
+        isRequired = true),
+
+      // Coordinate Descent Iterations
+      ScoptParameter[Int, Int](
+        GameTrainingDriver.coordinateDescentIterations,
+        usageText = "<value>",
+        isRequired = true),
+
+      // Normalization
+      ScoptParameter[NormalizationType, NormalizationType](
+        GameTrainingDriver.normalization,
+        usageText = "<type>",
+        additionalDocs = Seq(s"output modes: ${NormalizationType.values.map(_.toString).mkString(", ")}")),
+
+      // Data Summary Directory
+      ScoptParameter[Path, Path](
+        GameTrainingDriver.dataSummaryDirectory,
+        usageText = "<path>"),
+
+      // Tree Aggregate Depth
+      ScoptParameter[Int, Int](
+        GameTrainingDriver.treeAggregateDepth,
+        usageText = "<value>"),
+
+      // Hyper Parameter Tuning
+      ScoptParameter[HyperparameterTuningMode, HyperparameterTuningMode](
+        GameTrainingDriver.hyperParameterTuning,
+        usageText = "<type>",
+        additionalDocs = Seq(s"output modes: ${HyperparameterTuningMode.values.map(_.toString).mkString(", ")}")),
+
+      // Hyper Parameter Tuning Iterations
+      ScoptParameter[Int, Int](
+        GameTrainingDriver.hyperParameterTuningIter,
+        usageText = "<value>"),
+
+      // Hyper Parameter Tuning Range
+      ScoptParameter[String, DoubleRange](
+        GameTrainingDriver.hyperParameterTuningRange,
+        parse = ScoptParserHelpers.parseDoubleRange,
+        print = ScoptParserHelpers.doubleRangeToString,
+        usageText = "<start>-<end>"),
+
+      // Compute Variance
+      ScoptParameter[Boolean, Boolean](
+        GameTrainingDriver.computeVariance))
+
+  /**
+   * Parse command line arguments for GAME training into a [[ParamMap]].
+   *
+   * @param args [[Array]] of command line arguments
+   * @return An initialized [[ParamMap]]
+   */
+  def parseFromCommandLine(args: Array[String]): ParamMap = {
+
+    val parser = new OptionParser[ParamMap]("GAME-Training") {
+
+      private def optHelper[In, Out](scoptParameter: ScoptParameter[In, Out]): OptionDef[In, ParamMap] = {
+
+        implicit val read: Read[In] = scoptParameter.read
+
+        scoptParameter.toOptionDef(opt[In])
+      }
+
+      scoptGameTrainingParams.foreach(optHelper(_))
+    }
+
+    parser.parse(args, ParamMap.empty) match {
+      case Some(params) => params
+
+      case None =>
+        val errMsg = new StringBuilder()
+
+        for (i <- args.indices by 2) {
+          errMsg.append(s"${args(i)} ${args(i + 1)}\n")
+        }
```

Comment:
This would be a little more functional / readable / concise / hipstery / etc:

```scala
val errMsg = args
  .grouped(2)
  .map(_.mkString(" "))
  .mkString("\n")
```